### ISSUE TITLE
It would be nice to specify columns

### ISSUE THREAD

**Issue by nathans1309** [borisdj/EFCore.BulkExtensions#25]
One feature that would be useful is to specify the exact columns.  Any column not listed would be ignored by BulkExtensions which would allow the database to set the default values instead of EFCore.  For example, I am having an issue while upserting an Sql Server timestamp column where the default value is generated in the temp table and then fails when merging into the actual table because timestamps can't be explicitly inserted.

I would like to allow BulkConfig to override the list of columns to update.

**Comment by borisdj** [comment]
I understand the issue.
However proper solution for the situation you mentioned is actually to omit Properties with default value from inserting if their Columns have default value in Sql.
This is how regular EFCore Insert behaves so it's logical to make the same behavior.
I will see to implement this fix soon.

**Comment by nafberger** [comment]
@borisdj Can you please specify on how to contact you by email to continue our discussion. Big thanks

**Comment by borisdj** [comment]
Added feature that allows specifying exact columns to update.
Published NuGet 2.0.5

**Comment by nafberger** [comment]
I am trying the new thing but it throws an exception of:

"Dictionary key not found"

What am I doing wrong?

This is a snippet of my code:

`_context.BulkUpdate(productsToUpdate , new BulkConfig {  BulkCopyTimeout = 500, PropertiesToInclude = new List<string> { nameof(WalmartProduct.BuyBoxPrice) } });`

**Comment by borisdj** [comment]
You need to add **PrimaryKey** to list as well.
For example if PK is _WalmartProductId_ then it would be:
````
_context.BulkUpdate(
    productsToUpdate,
    new BulkConfig {
        BulkCopyTimeout = 500,
        PropertiesToInclude = new List<string> { nameof(WalmartProduct.WalmartProductId), nameof(WalmartProduct.BuyBoxPrice) }
    }
);
````

But now I see that I could implement this to be done automatically.
Add the moment just add this, and when I update the library I will write it here.

In addition regarding config **UpdateByProperties** purpose of it is when we want Update to be done not ny OK but by one other, or several combined columns, that should be effectively unique.
For example in your case if there would be another column say WalmartProductCode then:
````
_context.BulkUpdate(
                    productsToUpdate,
                    new BulkConfig {
                        BulkCopyTimeout = 500,
                        UpdateByProperties = new List<string> { nameof(WalmartProduct.WalmartProductCode) },
                        PropertiesToInclude = new List<string> { nameof(WalmartProduct.WalmartProductCode), nameof(WalmartProduct.BuyBoxPrice) }
                    }
                );
````

**Comment by borisdj** [comment]
@nafberger when using **PropertiesToInclude**, **PrimaryKey** column now(from 2.0.6 version) does not need to be explicitly added to **UpdateByProperties** list.
If not found in **PropertiesToInclude** list it will be automatically added in it from **UpdateByProperties** list.

**Comment by nafberger** [comment]
And that is even with a composite key. Also what happens if I do include the PK's? Thanks for your help

**Comment by borisdj** [comment]
You can include it, it will still work the same, it's just not required any more.
What it now does in the background is that it compares 2 lists: PrimaryKey list (composite key supported) and PropertiesToInclude list, and if it finds elements in first list that are not in the second it will add them there as well.

**Comment by nafberger** [comment]
Aha. Not sure if it's my problem, I upgraded to the latest version and it
threw an error when I didn't include the PK in the properties list, not
really sure if it's my problem or not.

**Comment by borisdj** [comment]
What was the error? the same as you would get on previous versions when PK is omitted from the list or something else now.

**Comment by nafberger** [comment]
I will try to post a screenshot

**Comment by borisdj** [comment]
You are right, it still throws an error.
I will have to make another fix, expect 2.0.7. soon.
